dbt_semantic_interfaces/validations/where_filters.py

```diff
@@ -0,0 +1,268 @@
+import traceback
+from typing import Generic, List, Sequence, Tuple
+
+from dbt_semantic_interfaces.call_parameter_sets import FilterCallParameterSets
+from dbt_semantic_interfaces.protocols import Metric, SemanticManifestT
+from dbt_semantic_interfaces.protocols.saved_query import SavedQuery
+from dbt_semantic_interfaces.references import MetricModelReference
+from dbt_semantic_interfaces.type_enums import TimeGranularity
+from dbt_semantic_interfaces.validations.validator_helpers import (
+    FileContext,
+    MetricContext,
+    SavedQueryContext,
+    SavedQueryElementType,
+    SemanticManifestValidationRule,
+    ValidationContext,
+    ValidationIssue,
+    ValidationWarning,
+    generate_exception_issue,
+    validate_safely,
+)
+
+
+class WhereFiltersAreParseableRule(SemanticManifestValidationRule[SemanticManifestT], Generic[SemanticManifestT]):
+    """Validates that all Metric WhereFilters are parseable."""
+
+    @staticmethod
+    def _validate_time_granularity_names_impl(
+        location_label_for_errors: str,
```

Comment:
This param seems kinda hacky - can we instead include 2 params? Something like this:
1. `element_name: str` (the name of the metric/saved query)
2. `object_type: SemanticManifestNodeType` - this would be either `SAVED_QUERY` or `METRIC`
Then you can build the location label as something like `f"{object_type.value} '{element_name}'"`